src/main/java/com/google/googleidentity/oauth2/token/JdbcOAuth2TokenService.java

```diff
@@ -361,8 +378,46 @@ public boolean revokeByRefreshToken(String refreshToken) {
       return false;
     }
 
-    revokeUserClientTokens(token.get().getUsername(), token.get().getClientId());
-
+    Connection conn = null;
+    PreparedStatement statement = null;
+    ResultSet result = null;
+    try {
+      conn = dataSource.getConnection();
+      conn.setAutoCommit(false);
+      String stmt = "DELETE FROM refresh_token WHERE refresh_token = ?;";
+      statement = conn.prepareStatement(stmt);
+      statement.setString(1, refreshToken);
+      statement.execute();
+      stmt = "DELETE FROM access_token WHERE refresh_token = ?;";
+      statement = conn.prepareStatement(stmt);
+      statement.setString(1, refreshToken);
+      statement.execute();
+      conn.commit();
+    } catch (SQLException exception) {
+      log.log(Level.INFO, "Revoke token error.", exception);
```

Comment:
We shall go through the code base again to double check we don't swallow the exceptions unless can correctly handle them.